doc/differentiation/lanczos_smoothing.qbk

```diff
@@ -0,0 +1,161 @@
+A basic usage is
+
+    std::vector<double> v(500);
+    // fill v with noisy data.
+    double spacing = 0.001;
+    using boost::math::differentiation::discrete_lanczos_derivative;
+    auto lanczos = discrete_lanczos_derivative(spacing);
+    // Compute dvdt at index 30:
+    double dvdt30 = lanczos(v, 30);
+    // Compute derivative of entire vector:
+    std::vector<double> dvdt = lanczos(v);
+
+Noise-suppressing second derivatives can also be computed.
+The syntax is as follows:
+
+    std::vector<double> v(500);
+    // fill v with noisy data.
+    auto lanczos = lanczos_derivative<double, 2>(spacing);
+    // evaluate second derivative at a point:
+    double d2vdt2 = lanczos(v, 18);
+    // evaluate second derivative of entire vector:
+    std::vector<double> d2vdt2 = lanczos(v);
+
+For memory conscious programmers, you can reuse the memory space for the derivative as follows:
+
+    std::vector<double> v(500);
+    std::vector<double> dvdt(500);
+    // . . . define spacing, create and instance of discrete_lanczos_derivative
+
+    // populate dvdt, perhaps in a loop:
+    lanczos(v, dvdt);
+
+
+If the data has variance \u03C3[super 2],
+then the variance of the computed derivative is roughly \u03C3[super 2]/p/[super 3] /n/[super -3] \u0394 /t/[super -2],
+i.e., it increases cubically with the approximation order /p/, linearly with the data variance,
+and decreases at the cube of the filter length /n/.
+In addition, we must not forget the discretization error which is /O/(\u0394 /t/[super /p/]).
+You can play around with the approximation order /p/ and the filter length /n/:
+
+    size_t n = 12;
+    size_t p = 2;
+    auto lanczos = lanczos_derivative(spacing, n, p);
+    double dvdt = lanczos(v, i);
+
+If /p=2n/, then the discrete Lanczos derivative is not smoothing:
+It reduces to the standard /2n+1/-point finite-difference formula.
+For /p>2n/, an assertion is hit as the filter is undefined.
+
+In our tests with AWGN, we have found the error decreases monotonically with /n/,
+as is expected from the theory discussed above.
+So the choice of /n/ is simple:
+As high as possible given your speed requirements (larger /n/ implies a longer filter and hence more compute),
+balanced against the danger of overfitting and averaging over non-stationarity.
+
+The choice of approximation order /p/ for a given /n/ is more difficult.
+If your signal is believed to be a polynomial,
+it does not make sense to set /p/ to larger than the polynomial degree-
+though it may be sensible to take /p/ less than this.
+
+For a sinusoidal signal contaminated with AWGN, we ran a few tests showing that for SNR = 1,
+p = n/8 gave the best results,
+for SNR = 10, p = n/7 was the best, and for SNR = 100, p = n/6 was the most reasonable choice.
+For SNR = 0.1, the method appears to be useless.
+The user is urged to use these results with caution-they have no theoretical backing and are extrapolated from a single case.
```

Comment:
Nitpick: There's a few places where you use a hyphen with no space around it which looks like a single hyphenated word to me rather than a separator between clauses.  Should be a colon maybe?  My grammatical abilities are severely limited so take my comments with a pinch of salt! ;)

Reply:
That indeed should be a colon. My grammatical training in English ended in the 6th grade, so I'm just going with what sounds best. (The other hyphen in `The filters are (regrettably) computed at runtime-the vast number of combinations` sounds good to me.)